### ISSUE TITLE
ndk 25c armv7-none-linux-androideabi33 error

### ISSUE THREAD

**Issue by calvin2021y** [Tencent/MMKV#1088]
### The language of MMKV

c 

### The version of MMKV

v1.2.16

### The platform of MMKV

Android

### The installation of MMKV

git clone

### What's the issue?

build with ndk 25c, target `armv7-none-linux-androideabi33`,  error form `openssl_aes-armv4.S`

```sh
clang -target armv7-none-linux-androideabi33 --sysroot=/android-ndk-r25/toolchains/llvm/prebuilt/darwin-x86_64/sysroot -DFORCE_POSIX  -x assembler-with-cpp -march=armv7a -D__ANDROID__ -O3 -DNDEBUG  -fPIC -o CMakeFiles/core.dir/aes/openssl/openssl_aes-armv4.S.o -c /mmkv/Core/aes/openssl/openssl_aes-armv4.S
```

link warn repeat a lot look like this:
```sh
ld.lld: warning: libmmkv.a(AESCrypt.cpp.o):(function mmkv::AESCrypt::statusBeforeDecrypt(void const*, void const*, unsigned int, mmkv::AESCryptStatus&): .text._ZN4mmkv8AESCrypt19statusBeforeDecryptEPKvS2_jRNS_14AESCryptStatusE+0x74): branch and link relocation: R_ARM_THM_CALL to non STT_FUNC symbol: openssl_aes_arm_set_decrypt_key interworking not performed; consider using directive '.type openssl_aes_arm_set_decrypt_key, %function' to give symbol type STT_FUNC if interworking between ARM and Thumb is required; AESCrypt.cpp:148 (Core/aes/AESCrypt.cpp:148)
```

### What's the log of MMKV when that happened?

```sh
AddressSanitizer:DEADLYSIGNAL
=================================================================
==23283==ERROR: AddressSanitizer: SEGV on unknown address 0x1a30020b (pc 0x0720115a bp 0xffee7151 sp 0xffee6e70 T0)
==23283==The signal is caused by a WRITE memory access.
    #0 0x720115a in AES_Te AESCrypt.cpp
    #1 0x7106ad0 in __sanitizer::BufferedStackTrace::UnwindImpl(unsigned int, unsigned int, void*, bool, unsigned int) out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/asan_stack.cpp
    #2 0x70a17e0 in Unwind out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/../sanitizer_common/sanitizer_stacktrace.h:131:5
    #3 0x70a17e0 in asan::OnStackUnwind(sanitizer::SignalContext const&, void const*, __sanitizer::BufferedStackTrace*) out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/asan_errors.cpp:38:10
    #4 0x7131748 in ReportDeadlySignalImpl out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_report.cpp:229:3
    #5 0x7131748 in sanitizer::ReportDeadlySignal(sanitizer::SignalContext const&, unsigned int, void (*)(sanitizer::SignalContext const&, void const*, sanitizer::BufferedStackTrace*), void const*) out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/sanitizer_common/sanitizer_symbolizer_report.cpp:243:5
    #6 0x70fdcdc in __asan::ScopedInErrorReport::~ScopedInErrorReport() out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/asan_report.cpp:141:50
    #7 0x70fdadc in asan::ReportDeadlySignal(sanitizer::SignalContext const&) out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/asan_report.cpp:218:1
    #8 0x70fca04 in __asan::AsanOnDeadlySignal(int, void*, void*) out/lib/compiler-rt-arm-ndk-cxx/out/llvm-project/compiler-rt/lib/asan/asan_posix.cpp:39:3
    #9 0xf32747d0  (/apex/com.android.runtime/lib/bionic/libc.so+0x347d0) (BuildId: f151d05f5e13d70c25120e896656ffe1)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV AESCrypt.cpp in AES_Te
==23283==ABORTING
```

**Comment by calvin2021y** [comment]
test this patch fix it:

```diff
diff --git a/Core/aes/openssl/openssl_aes-armv4.S b/Core/aes/openssl/openssl_aes-armv4.S
index 1f8c07f..6199fe8 100644
--- a/Core/aes/openssl/openssl_aes-armv4.S
+++ b/Core/aes/openssl/openssl_aes-armv4.S
@@ -161,6 +161,7 @@ AES_Te:
 
 /* void openssl_aes_arm_encrypt(const uint8_t *in, uint8_t *out, const AES_KEY *key) {
 */
+.type openssl_aes_arm_encrypt, %function
 #ifndef __linux__
 .globl	_openssl_aes_arm_encrypt
 #ifdef __thumb2__
@@ -421,7 +422,7 @@ Lenc_loop:
 	sub	r10,r10,#2
 	ldr	pc,[sp],#4		// pop and return
 
-
+.type openssl_aes_arm_set_encrypt_key, %function
 #ifndef __linux__
 .globl	_openssl_aes_arm_set_encrypt_key
 #ifdef __thumb2__
@@ -735,6 +736,7 @@ Labrt:
 .word	0xe12fff1e			// interoperable with Thumb ISA:-)
 #endif
 
+.type     openssl_aes_arm_set_decrypt_key, %function
 #ifndef __linux__
 .globl    _openssl_aes_arm_set_decrypt_key
 #ifdef __thumb2__
@@ -955,6 +957,7 @@ AES_Td:
 
 
 // void AES_decrypt(const uint8_t *in, uint8_t *out, const AES_KEY *key) {
+.type openssl_aes_arm_decrypt, %function
 #ifndef __linux__
 .globl    _openssl_aes_arm_decrypt
 #ifdef __thumb2__
```

**Comment by lingol** [comment]
Noted.

**Comment by lingol** [comment]
Fixed in v1.3.0.
